automata/pda/pda.py

```diff
@@ -2,18 +2,34 @@
 """Classes and methods for working with all pushdown automata."""
 
 import abc
-from typing import AbstractSet, Literal
+import os
+import pathlib
+import random
+import uuid
+from collections import defaultdict
+from typing import AbstractSet, Generator, List, Literal, Optional, Tuple, Union
 
 import automata.base.exceptions as exceptions
 import automata.pda.exceptions as pda_exceptions
 from automata.base.automaton import Automaton, AutomatonStateT, AutomatonTransitionsT
 from automata.pda.configuration import PDAConfiguration
 from automata.pda.stack import PDAStack
 
+# Optional imports for use with visual functionality
+try:
+    import coloraide
+    import pygraphviz as pgv
+except ImportError:
+    _visual_imports = False
+else:
+    _visual_imports = True
+
 PDAStateT = AutomatonStateT
 PDATransitionsT = AutomatonTransitionsT
 PDAAcceptanceModeT = Literal["final_state", "empty_stack", "both"]
 
+LayoutMethod = Literal["neato", "dot", "twopi", "circo", "fdp", "nop"]
```

Comment:
Could this type be put in a shared file somewhere, just so we don't have to have this duplicated?